### ISSUE TITLE
exception when viewing plot for a databroker entry 

### ISSUE THREAD

**Issue by canismarko** [bluesky/tiled#371]
Hi, all.

I installed tiled recently and I can visit the site and navigate to my data, but I'm not able to actually view data.

When I go to a structure I want to view, I just see the page with a placeholder for the plot:
![Screenshot from 2022-11-29 15-40-55](https://user-images.githubusercontent.com/2363800/204654061-ce0cde94-7a19-4500-bfdc-051117417b47.png)

The console has a bunch of exceptions (https://gist.github.com/canismarko/5bb24f92326cb21e9e718f8c15ecd617) where the last one ends:

    File "/home/beams0/S25STAFF/micromamba/envs/tiled/lib/python3.10/site-packages/tiled/server/core.py", line 298, in <listcomp>
         for spec in [spec.name for spec in specs] + [structure_family]:
    AttributeError: 'str' object has no attribute 'name'

I dropped into pdb and ``specs``​ is a list of strings (["x_array_data"]or something), and so of course spec.name raises AttributeError. I wasn't able to figure out why it has that value though (it's in an async call so the call stack is hard to sort out).

Here is the configuration I'm using: https://github.com/spc-group/tiled/blob/main/config.yml

And the script I'm using to start tiled (the actual command is at the end): https://github.com/spc-group/tiled/blob/main/start-tiled.sh

I'm using tiled==0.1.0a80, python==3.10, on a linux computer (RHEL 8). I've tried in both firefox 91.11.0esr and chrome 105.0.5195.102.

**Comment by danielballan** [comment]
Thanks for this very clear report @canismarko. What version of databroker are you running?

In https://github.com/bluesky/tiled/pull/356, released in tiled v0.1.0a79, we changed `spec` from a simple string to a dict. We updated databroker in https://github.com/bluesky/databroker/pull/744, released in databroker v2.0.0b12, to align with this change.

My guess is that you have a too-old version of databroker. Does upgrading fix it?

Thanks for your patience with the bumps. We made couple backward-incompatible changes over the last month, accounting for what we've learned seeing tiled used in practice.
